src/Context/Annotation/ContextDefinition.php

```diff
@@ -38,6 +38,8 @@ public function __construct(array $values) {
     // Default to Rules context definition class.
     $values['class'] = isset($values['class']) ? $values['class'] : '\Drupal\rules\Context\ContextDefinition';
     parent::__construct($values);
+    $additional_values = $values + ['allow_null' => FALSE];
```

Comment:
That works, but could seems to be simpler if we'd just switch to the ContextDefinition::createFromArray(). Also, I guess it's better to centralize usual construction logic and bake class check etc. in there also.